demos/common/python/openvino/model_zoo/model_api/README.md

```diff
@@ -0,0 +1,145 @@
+
+Currently, `OpenvinoAdapter` and `OVMSAdapter` are supported.
+
+#### OpenVINO executor
+
+`OpenvinoAdapter` hides the OpenVINO™ toolkit API, which allows Model API wrappers launching with models represented in Intermediate Representation (IR) format.
+It accepts a path to either `xml` model file or `onnx` model file.
+
+For OpenVINO executor employment, you need to install the requirements:
+```sh
+pip install <omz_dir>/demos/common/python/requirements_openvino.txt
+```
+
+#### OpenVINO Model Server executor
+
+`OVMSAdapter` hides the OpenVINO Model Server python client API, which allows Model API wrappers launching with models served by OVMS.
+
+Refer to __[`OVMSAdapter`](adapters/ovms_adapter.md)__ to learn about running demos with OVMS.
+
+For OpenVINO Model Server executor employment, you need to install the requirements:
+```sh
+pip install <omz_dir>/demos/common/python/requirements_ovms.txt
+```
+
+## Model API Pipelines
+
+Model API Pipelines represent the high-level wrappers upon the input data and accessing model results management.
+It performs the data submission for model inference, verification of inference status, whether the result is ready or not, and results accessing.
+
+The `AsyncPipeline` is available, which handles the asynchronous execution of a single model.
+
+## Ready-to-use Model API solutions
+
+To apply Model API wrappers in custom applications, learn the provided example of common scenario of how to use Python* Model API.
+
+ In the example, the SSD architecture is used to predict bounding boxes on input image `"sample.png"`. The model execution is produced by `OpenvinoAdapter`, therefore we submit the path to the model's `xml` file. The model is loaded on a CPU device inside the adapter.
+
+Once the SSD model wrapper instance is created, we get the predictions by the model in one line: `ssd_model(input_data)` - the wrapper performs the preprocess method, synchronous inference on OpenVINO side and postprocess method.
+
+```python
+import cv2
+from openvino.model_zoo.model_api.models import SSD
+from openvino.model_zoo.model_api.adapters import OpenvinoAdapter, create_core
+
+
+# a helper function for bboxes visualization
+def draw_detections(image, detections):
+    for detection in detections:
+        class_id = int(detection.id)
+        color = (255, 0, 0)
+        det_label = '#{}'.format(class_id)
+        xmin, ymin, xmax, ymax = detection.get_coords()
+        cv2.rectangle(image, (xmin, ymin), (xmax, ymax), color, 2)
+        cv2.putText(image, '{} {:.1%}'.format(det_label, detection.score),
+                    (xmin, ymin - 7), cv2.FONT_HERSHEY_COMPLEX, 0.6, color, 1)
+    return image
+
+
+def main():
+    input_data = cv2.imread("sample.png")
+    model_path = "public/mobilenet-ssd/FP32/mobilenet-ssd.xml"
+
+    model_adapter = OpenvinoAdapter(create_core(), model_path, device="CPU")
+    ssd_model = SSD(model_adapter, preload=True)
+
+    results = ssd_model(input_data)
```

Comment:
I think these lines for demonstration how api should be used is enough, also you can say that you can find a lot of examples in demos. It will be great if we can have list of model api supported demos also

Reply:
Added the list of such demos.

> I think these lines for demonstration how api should be used is enough, also you can say that you can find a lot of examples in demos  

The reference to OMZ demos is good, but I found useful also to add the visualization to this snippet. Someone can just take this code, launch and see clearly how the Model API works. With only these lines it's not obvious, what the format of `results` is (as we remember from demos, it is the list of `Detection` structure instances).

So, I think this example for sync inference should be self-contained enough and representative as possible. For the more complex usages, like async inference, there is the recommendation to refer to OMZ demos

Reply:
I confused by main function and other stuff which make example more similar to ready application, my opinion that we should just provide api with comments which line used for what (where model loaded, where data provided, where results should be visualized and so on) without wrapping to application

Reply:
BTW, what about submitting jupiter notebook to https://github.com/openvinotoolkit/openvino_notebooks
with demonstration how to work with model api? Is it planned for future?

Reply:
Updated example according to your suggestions.

The `openvino_notebooks` team waits then the package becomes the part of openvino-dev package as I remember. Possibly the notebook with model API with demonstration will be useful, we should discuss it with notebooks maintainers.